src/plugins/kconfig/file_utility.hpp

```diff
@@ -0,0 +1,129 @@
+#ifndef ELEKTRA_FILEUTILITY_HPP
+#define ELEKTRA_FILEUTILITY_HPP
+
+#include "kconfig_parser_exception.hpp"
+#include <memory>
+#include <sstream>
+
+namespace kconfig
+{
+class FileUtility
+{
+private:
+	/* This file is the file stream that we want to parse  */
+	std::unique_ptr<std::istream> file;
+	/* This stringBuffer is used when reading strings from file so that we don't initialize a stringstream each time */
+	std::stringstream stringBuffer;
+	/* This int is used to save the current line number for the purpose of better error messages */
+	int currentLine;
+	/* This string is used to save the filename for the purpose of better error messages */
+	std::string filename;
+
+public:
+	/**
+	 * @brief This constructor is used to abstract the file stream into the methods that are specific to parsing a kconfig configuration
+	 * file
+	 * @param filename This string contains the absolute/relative path to the file
+	 */
+	explicit FileUtility (std::string filenameParam, std::unique_ptr<std::istream> fileParam);
+
+	/**
+	 * \copydoc std::istream::peek()
+	 */
+	char peekNextChar ();
+
+	/**
+	 * @brief This method is used to check if the next character in the buffer is end of file
+	 *
+	 * @return true if the next character is end of file, false otherwise
```

Comment:
If you want you can use the Doxygen command [`retval`](http://www.doxygen.nl/manual/commands.html#cmdretval) to specify the different return values.

Reply:
Yes, it should be two retval (true and false) then.